client/components/rich_text/components/paragraph.coffee

```diff
@@ -60,6 +60,11 @@ module.exports = React.createClass
           new CompositeDecorator Config.decorators(@hasLinks())
         )
 
+  componentDidUpdate: (prevProps) ->
+    if @props.html != prevProps.html && @props.html != @state.html
+      # re-initialize on drag/drop
+      @componentDidMount()
```

Comment:
I don't _think_ there's an issue with calling directly with this, but maybe it'd feel more natural to move the work in `componentDidMount` to a `setParagraphState` and call the method. Not a blocker, feel free to follow up on it if you wish.